tests/Sinch.Tests/Mailgun/MailgunClientTests.cs

```diff
@@ -0,0 +1,28 @@
+using System;
+using FluentAssertions;
+using Sinch.Mailgun;
+using Xunit;
+
+namespace Sinch.Tests.Mailgun
+{
+    public class MailgunClientTests
+    {
+        [Fact]
+        public void InitMailgunClient()
+        {
+            var mailgun = new SinchClient(default, default, default).Mailgun("apikey", MailgunRegion.Eu);
```

Comment:
I know it's following the same design as Verification and Voice which are not using the same kind of credentials, but it still seems very weird to have to initialize a SinchClient with a triplet of undefined parameters to be able to access to the Mailgun constructor with its own parameters; knowing that on top of that, the overrides for the URL should be part of the optional fourth parameter of the SinchClient.
Do you think it would be valuable to review this design?

Reply:
I'm not sure to understand what you want to do, but let's go ahead and give it a try.